### PR TITLE
build: neovim derivation with rocks.nvim for manual testing with nix

### PR DESCRIPTION
**Pull request by mrcjkb**
Currently, you have to call `:lua require('rocks').setup()` for the plugin to initialize.

### FIXMEs:

- [x] We need to get rid of the `setup` function (separate PR)
- [x] I had to add `toml` and `toml-edit` to the `LUA_CPATH` via `wrapperArgs`(@teto I thought `buildNeovimPlugin` adds the dependencies?).
- [ ] Haven't been able to test `:Rocks install` yet.

### CODE REVIEW

**Comment by teto** [comment]
>  I had to add toml and toml-edit to the LUA_CPATH via wrapperArgs(@teto I thought buildNeovimPlugin adds the dependencies?).

I think that's a bug in nixpkgs, the hook may not work correctly

File: nix/plugin-overlay.nix [hunk 1]
```diff
@@ -1,9 +1,68 @@
...
+      plugins = with final.vimPlugins; [
+        rocks-nvim
+      ];
+    };
+    runtimeDeps = with final; [
+      lua5_1
+      luarocks
+    ];
+  in
+    final.wrapNeovimUnstable final.neovim-nightly (neovimConfig
```
Comment (teto) [inline nix/plugin-overlay.nix:50]: pretty cool. Hopefully ithe interface will become easier after 23.11. I prefer to keep the wrapperArgs as a list and was thinking of making it mandatory in next nixpkgs (but I can't justify it yet so maybe wont).

File: nix/plugin-overlay.nix [hunk 2]
```diff
@@ -1,9 +1,68 @@
...
+      // {
+        wrapperArgs =
+          lib.escapeShellArgs neovimConfig.wrapperArgs
+          + " "
+          + ''--set NVIM_APPNAME "nvimrocks"''
```
Comment (teto) [inline nix/plugin-overlay.nix:55]: pretty cool usage of NVIM_APPNAME

**Review by teto — APPROVED** [review]

File: nix/plugin-overlay.nix (hunk 2 above)
Comment (mrcjkb) [reply nix/plugin-overlay.nix:55]: It's a really nice feature, especially for Nix...

**Pull request merged**